lib/season_stats.rb

```diff
@@ -0,0 +1,91 @@
+module SeasonStats
+
+  def season_with_most_games
+    highest_count = count_of_games_by_season.values.max
+    count_of_games_by_season.key(highest_count).to_i
+  end
+
+  def season_with_fewest_games
+    lowest_count = count_of_games_by_season.values.min
+    count_of_games_by_season.key(lowest_count).to_i
+  end
+
+  def count_of_games_by_season
+    game_count_by_season = {}
+    all_games_by_season.each do |season, games|
+      game_count_by_season[season] = games.count
+    end
+    return game_count_by_season
+  end
+
+  def group_games_by_season_type(type, games = @games)
```

Comment:
@aprildagonese  This replaces the games_by_season_type method you referenced earlier. Instead of having to create a hash, we just pass the type as an argument. So we pass "P" or "R" and an array of games that I defaulted to ALL games.